scripts/proposals/reserveRestore.ts

```diff
@@ -0,0 +1,413 @@
+    "registerScheme(address,bytes32,bytes4,address)", // give upgrade contract permissions
+    "upgrade(address)",
+    "unpause()"
+  ];
+
+  const proposalFunctionInputs = [
+    ethers.utils.defaultAbiCoder.encode(["bool"], [true]),
+    ethers.utils.defaultAbiCoder.encode(["bool"], [true]),
+    ethers.utils.defaultAbiCoder.encode(["uint256"], [0]),
+    ethers.utils.defaultAbiCoder.encode(["uint256", "uint256"], [999711382710978, 1e15]),
+    ethers.utils.defaultAbiCoder.encode(["uint256"], [0]),
+    ethers.utils.defaultAbiCoder.encode(["address"], ["0xD17652350Cfd2A37bA2f947C910987a3B1A1c60d"]),
+    ethers.utils.defaultAbiCoder.encode(["address"], ["0xeC577447D314cf1e443e9f4488216651450DBE7c"]),
+    ethers.utils.defaultAbiCoder.encode(["address"], ["0x6738fA889fF31F82d9Fe8862ec025dbE318f3Fde"]),
+    ethers.utils.defaultAbiCoder.encode(["address", "address"], [release.GoodDollar, release.MpbBridge]),
+    ethers.utils.defaultAbiCoder.encode(["address"], [reserveImpl.address]),
+    ethers.utils.defaultAbiCoder.encode(["address"], [goodFundManagerImpl.address]),
+    ethers.utils.defaultAbiCoder.encode(["address"], [exchangeHelperImpl.address]),
+    ethers.utils.defaultAbiCoder.encode(["address"], [distHelperImplt.address]),
+    ethers.utils.defaultAbiCoder.encode(["address"], [stakersDistImpl.address]),
+    ethers.utils.defaultAbiCoder.encode(["address"], [marketMakerImpl.address]),
+    ethers.utils.defaultAbiCoder.encode(["address"], [govImpl.address]),
+    "0x", //setAddresses
+    ethers.utils.defaultAbiCoder.encode(
+      ["address", "bytes32", "bytes4", "address"],
+      [
+        upgradeImpl.address, //scheme
+        ethers.constants.HashZero, //paramshash
+        "0x000000f1", //permissions - minimal
+        release.Avatar
+      ]
+    ),
+    ethers.utils.defaultAbiCoder.encode(["address"], [funder]),
+    "0x"
+  ];
+
+  if (isProduction) {
+    await executeViaSafe(
+      proposalContracts,
+      proposalEthValues,
+      proposalFunctionSignatures,
+      proposalFunctionInputs,
+      protocolSettings.guardiansSafe,
+      "mainnet"
+    );
+  } else {
+    //simulation or dev envs
+    await executeViaGuardian(
+      proposalContracts,
+      proposalEthValues,
+      proposalFunctionSignatures,
+      proposalFunctionInputs,
+      guardian,
+      networkName
+    );
+  }
+
+  if (isSimulation) {
+    await mainnetPostChecks()
+  }
+};
+
+const mainnetPostChecks = async () => {
+  networkName = "production-mainnet";
+  let release: { [key: string]: any } = dao[networkName];
+
+  let [root, ...signers] = await ethers.getSigners();
+  const gd = await ethers.getContractAt("IGoodDollar", release.GoodDollar);
+
+  const locked = await ethers.getImpersonatedSigner("0xeC577447D314cf1e443e9f4488216651450DBE7c");
+  const tx = await gd
+    .connect(locked)
+    .burn("10", { maxFeePerGas: 30e9, maxPriorityFeePerGas: 1e9, gasLimit: 200000 })
+    .then(_ => _.wait())
+    .then(_ => _.status)
+    .catch(e => e);
```

Comment:
yes it could.
it could also succeed and not throw and error, both cases should not happen.
but again this is for simulation where I looked at the result reason.

this is really not important. I understand this is best practice, but again best practices are recommendations not rules set in stone, it is an overkill for this case.

Reply:
because its an overkill it doesnt add any value. the result is needed to be evaluated visually, if it succeeds or not and if not what is the error message.
an actual value added change would be to programmatically check the right error is thrown and throw an error or show a big message if it succeeds or the wrong error is recieved. but again all of this is an overkill for the purpose of the code.
the code does what it is supposed to do, which is showing the error to the developer.